### ISSUE TITLE
BlueToothLE - there should be an event for incoming data

### ISSUE THREAD

**Issue by kezb765** [thunkable/thunkable-issues#380]
Some months ago I built an app with BlueTooth Low Energy (BLE) under Thunkable classic. I was able to scan my environment, connect to a device, send data and respond to incoming data. In ThunkableX I am missing an event that fires when BLE data comes in, because it is not always possible (or wanted) to permanently scan the input queue.

**Describe the solution you'd like**
Add a block for receiving data as an event (a block with "golden" color, not purple).
That way it could be possible to write the following blocks, for example to receive data from a BLE device without waiting for it:
--> 'when BlueToothBLE receive: do "from Label1 set Text to receivedData"' etc.

**Describe alternatives you've considered**
Use Thunkable classic or other appy-builders for Android. For iOS there is no alternative.

**Which platform are you asking for?**
Thunkable ✕ (x.thunkable.com)

**Additional context**

**Comment by gre4twh1te** [comment]
true but not only is it missing an event for incoming data but it also needs to support HEX values for many BLE devices and we need an "is connected" block as well as "disconnect device xy"

**Comment by kezb765** [comment]
Yes, absolutely right. Compared with AppyBuilder (similar to Thunkable classic) we have in ThunkableX an incomplete framework for BlueTooth BLE.
